lib/src/main/java/org/epics/archiverappliance/retrieval/client/RawDataRetrieval.java

```diff
@@ -71,13 +78,15 @@ public final GenMsgIterator getDataForPVs(
         String getURL = buf.toString();
         logger.info("URL to fetch data is " + getURL);
         try {
-            URL url = new URL(getURL);
-            var data = Request.get(url.toURI()).execute().returnContent();
-
-            BufferedInputStream is = new BufferedInputStream(data.asStream());
-
-            if (is.available() <= 0) return null;
```

Comment:
Maybe keep the check that the body isn't empty as well.

Reply:
Makes sense. Pushed changes with this back in. Thx for the heads up.